dicom_utils/container/input.py

```diff
@@ -156,17 +155,16 @@ def __init__(
             self.records = records
         self.groups = [GROUP_REGISTRY.get(g) for g in groups]
         self.namers = [NAME_REGISTRY.get(n)() for n in namers]
+        if not groups:
+            # Is there a use case for Input where a grouping operation wouldn't be used?
+            raise ValueError("`groups` cannot be empty")
         if len(self.namers) != len(self.groups):
             raise ValueError("Number of namers {namers} should match number of groups {groups}")
 
         # scan sources and build a RecordCollection with every valid file found
-        sources = [Path(sources)] if isinstance(sources, PathLike) else [Path(p) for p in sources]
-        scan_source = partial(
-            RecordCollection.from_dir, record_types=self.records, helpers=helpers, filters=filters, **kwargs
-        )
-        collection = reduce(
-            lambda c1, c2: c1.union(c2),
-            (scan_source(s) for s in sources),
+        sources = [Path(sources)] if isinstance(sources, PathLike) else (Path(p) for p in sources)
```

Comment:
Why not make sources of type `List[Path]` in either case?

Reply:
Since `sources` could be very long (potentially 100k-1M), I kept it as an iterator. Using a list may add a delay with no progress bar to indicate what is happening.

Reply:
Makes sense. Maybe this isn't worth worrying about, but what about changing this line so that `sources` is always of type `Generator`? I.e.
```
sources = (Path(p) for p in [sources] if isinstance(sources, PathLike) else sources)
```